File: scala_study/src/main/scala/leetcode/getIntersectionNode.scala
package leetcode

class ListNode(_x: Int, _next: ListNode) {
  var next: ListNode= _next
  var x: Int = _x

}

object Solution{
  def getIntersectionNode(headA:ListNode,headB:ListNode):ListNode={
    if (headA == null || headB == null) return null
    var pa = headA
    var pb = headB
    while(pa != pb){
      if (pa ==null) pa = headB else pa=pa.next
      if (pb ==null) pb = headA else pb=pb.next
    }
   pb
  }
}

File: JavaLearn/src/com/leetcode/GetIntersectionNode.java
package com.leetcode;

public class GetIntersectionNode {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA;
        ListNode b = headB;

        if (a == null || b == null){
            return null;
        }

        while (a!=b){

//            if (a == null && b != null){
//                a = headB;
//            }
//            if (a!=null && b == null){
//                b = headA;
//            }
//
//            a=a.next;
//            b=b.next;

            a = a == null ? a = headB : a.next;
            b = b == null ? b = headA : b.next;
        }

        return a;

    }
}
